### DISCUSSION TITLE
proper way to grow/shrink a mesh without scaling plotter?

### DISCUSSION THREAD

**Discussion by AndrewAnnex** [pyvista/pyvista#1545] [Q&A] [answered]
Is there a way to shrink/grow a mesh using the api? I attempted to use shrink and transform with a custom transformation matrix but it seems to scale things relative to the origin rather than the centroid of the mesh and scale didn't seem to do anything at all (and it'
is inconsistent with the api as it doesn't accept an inplace flag).

**Comment by AndrewAnnex** [comment, marked as answer]
while thinking about this more, I managed I think to make a routine to do this
```
def shrink2(m, s):
    c = np.array(m.center)
    mm = m.copy(deep=True)
    mm.translate(-c)
    mm.transform(np.array([
        [s,0,0,0],
        [0,s,0,0],
        [0,0,s,0],
        [0,0,0,1]
    ]))
    mm.translate(c)
    return mm
```
but I am no 100% sure about the transformation matrix

**Reply by akaszynski** [reply]
This works, and so would:

```py
m.points[:] = (m.points - m.center)*s + m.center
```
Where `m` is the mesh and `s` is the shrink value.
